### PR TITLE
Fix multiproc metrics in no_trainer examples

### PR DESCRIPTION
**Pull request by muellerzr**
# Fixes a bug in the `no_trainer` scripts involving metric evaluation

## What does this add?

Fixes up the metric evaluation in most of the `no_trainer` scripts when multi-processing is involved. The final batch gets duplicated in those situations, so the metrics are slightly higher (or lower) than their actual values.

## Who is it for?

Users of the `no_trainer` scripts

### CODE REVIEW

File: examples/pytorch/image-classification/run_image_classification_no_trainer.py [hunk 1]
```diff
@@ -457,12 +457,21 @@ def collate_fn(examples):
                 break
 
         model.eval()
+        samples_seen = 0
         for step, batch in enumerate(eval_dataloader):
             outputs = model(**batch)
             predictions = outputs.logits.argmax(dim=-1)
+            predictions, references = accelerator.gather((predictions, batch["labels"]))
+            # If we are in a multiprocessing environment, the last batch has duplicates
```
Comment (sgugger) [inline examples/pytorch/image-classification/run_image_classification_no_trainer.py:465]:
```suggestion
            # If we are in a multiprocess environment, the last batch has duplicates
```

**Review by sgugger — APPROVED** [review]
LGTM, with one nit to propagate across examples :-)
Thanks!
